### PR TITLE
[CUDA] Enable XQA decode for GroupQueryAttention with attention sink

### PR DESCRIPTION
**Pull request by tianleiwu**
## Description

This PR enables the XQA decode kernel for the CUDA `GroupQueryAttention` (GQA) operator when an
attention-sink input (`head_sink`) is present, the common pattern in GPT-OSS style decode models.
The sink is treated as a smooth-softmax term, and a `PrePack` step converts a constant `head_sink`
initializer to a cached FP32 buffer once at session init to avoid a per-step conversion. XQA now
turns on by default for the `head_sink` decode path while preserving the existing
`ORT_ENABLE_XQA` opt-in/opt-out semantics for all other non-quantized cases.

### Kernel: XQA dispatch and head_sink handling

| File | Change |
|------|--------|
| `onnxruntime/contrib_ops/cuda/bert/group_query_attention.cc` | Add `PrePack` that caches a constant `head_sink` initializer as FP32 (`xqa_head_sink_`); allow XQA when `head_sink` is present (smooth-softmax via attention sink); default XQA on for the `head_sink` decode path; add `xqa_force_disabled_` so an explicit `ORT_ENABLE_XQA=0` always wins; reserve per-launch FP32 scratch when `head_sink` is dynamic (not prepacked). |
| `onnxruntime/contrib_ops/cuda/bert/group_query_attention.h` | Add `PrePack` declaration and members: `xqa_head_sink_`, `xqa_head_sink_count_`, `xqa_force_disabled_`. |
| `onnxruntime/contrib_ops/cuda/bert/group_query_attention_impl.cu` / `.h` | Add `LaunchConvertHeadSinkToFloat` to convert FP16/BF16 `head_sink` to FP32 for XQA. |
| `onnxruntime/contrib_ops/cuda/bert/attention_data.h` | Add `xqa_head_sink` (FP32 sink pointer) and `xqa_head_sink_needs_conversion` to `GroupQueryAttentionData`. |
| `onnxruntime/contrib_ops/cuda/bert/attention_kernel_options.{cc,h}` | Add `use_xqa` to the debug-info print so `SdpaKernel=XQA` is reported. |

### XQA loaders: attention-sink plumbing

- `onnxruntime/contrib_ops/cuda/bert/xqa/xqa_loader*.{cu,cuh,h}` and `xqa_impl_gen.cuh` — thread the
  FP32 attention-sink pointer through the FP16/BF16 (and int8/fp8 KV) XQA loader entry points.

### Tests and docs

| File | Change |
|------|--------|
| `onnxruntime/test/python/transformers/test_gqa.py` | Add `TestXQAHeadSinkParity` with runtime and PrePack (`head_sink` as initializer) parity cases; add `has_xqa()` skip guard; `setUp` clears `ORT_ENABLE_XQA` to exercise the real default-on behavior. |
| `onnxruntime/test/python/transformers/gqa_test_helper.py` | Support `head_sink` plumbing for the new tests. |
| `onnxruntime/test/python/transformers/profile_gqa.py` | Minor `head_sink` profiling support. |
| `docs/contrib_ops/gqa.md` | New document describing the GQA operator, inputs/attributes, and XQA selection defaults (quantized on; non-quantized `head_sink` on; otherwise opt-in via `ORT_ENABLE_XQA`; `ORT_ENABLE_XQA=0` disables). |

## Testing

- Run the XQA parity suites on an Ampere+ GPU:
  ```
  cd onnxruntime/test/python/transformers
  python -m pytest test_gqa.py -k "TestXQAHeadSinkParity or TestXQAQuantizedParity" -q
  ```
  All 224 cases pass on H200 (SM90). Tests skip automatically on devices without XQA support.
- Kernel selection was verified via `ORT_ENABLE_ATTENTION_KERNEL_DEBUG_INFO=1`:
  - `head_sink` present, env unset → `SdpaKernel=XQA`.
  - `head_sink` present, `ORT_ENABLE_XQA=0` → falls back to `SdpaKernel=FLASH_ATTENTION` (parity still passes).
  - Non-`head_sink` decode is unchanged (still Flash / cuDNN).
- Backward compatibility: Flash/fallback paths keep the original FP16/BF16 `head_sink`, so XQA is a
  performance default, not a correctness requirement; `ORT_ENABLE_XQA=1/0` semantics are preserved.

## Motivation and Context

GPT-OSS style decode models use a per-head attention sink. Routing these decode steps through XQA
improves decode latency, and prepacking the constant sink to FP32 removes a per-step conversion.
This PR targets `main` but depends on PR #29161 (FlashDecode split planning for local-window GQA),
which should merge first. Until then this PR's diff also includes the #29161 commit; once #29161
lands on `main`, GitHub's merge-base will drop it and this PR's diff will contain only the XQA change.

## Checklist

- [x] Tests added/updated
- [x] Documentation updated (`docs/contrib_ops/gqa.md`)
- [x] No breaking changes (existing `ORT_ENABLE_XQA` semantics preserved)
- [x] CI passes

### CODE REVIEW

File: onnxruntime/test/python/transformers/test_gqa.py [hunk 1]
```diff
@@ -2335,6 +2367,120 @@ def test_xqa_quantized_parity(self, name, config, torch_type, ort_type):
         )
 
 
+def gqa_xqa_head_sink_test_cases():
+    # Non-quantized global decode with a head_sink (attention sink) input.
+    # These configs exercise the XQA attention-sink path added for GPT-OSS style models:
+    # seq_len=1, shared KV buffer, no softcap, no local window, head_size in {64, 128, 256},
+    # and 64 % group_size == 0.
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply onnxruntime/test/python/transformers/test_gqa.py:2374]: Fixed: updated the comment to `head_size in {64, 128}` to match the loop, which only yields 64 and 128.

File: docs/contrib_ops/gqa.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+## 9. Testing
+
+CUDA parity tests live in
+[onnxruntime/test/python/transformers/test_gqa.py](../../onnxruntime/test/python/transformers/test_gqa.py):
+
+- `TestXQAQuantizedParity` — XQA per-tensor int8 quantized decode parity.
+- `TestXQAHeadSinkParity` — non-quantized XQA decode parity with a `head_sink` (attention sink) input.
+
+Both classes set `ORT_ENABLE_XQA=1` so the XQA path is exercised, and compare against a PyTorch
+reference (`attention_ref` with `smooth_softmax_ref`).
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply docs/contrib_ops/gqa.md]: Fixed: the testing section now states that `TestXQAQuantizedParity` sets `ORT_ENABLE_XQA=1` while `TestXQAHeadSinkParity` clears `ORT_ENABLE_XQA` to validate the default-on head_sink behavior.

File: onnxruntime/test/python/transformers/profile_gqa.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  nsys profile -o gqa_int8 --export=sqlite python profile_gqa.py --mode int8 --warmup 5 --repeat 10
+  python parse_nsys.py gqa_int8.sqlite
+"""
+
+import argparse
+import time
+
+import torch
+from test_sparse_attention import GroupQueryAttentionConfig, OrtGroupQueryAttention
+
```
[the remark this answers is not in the document]
Comment (tianleiwu) [reply onnxruntime/test/python/transformers/profile_gqa.py:35]: Fixed: `profile_gqa.py` now imports `GroupQueryAttentionConfig`/`OrtGroupQueryAttention` directly from `gqa_test_helper` using the same try/except `sys.path` fallback pattern as `test_sparse_attention.py`.

File: docs/contrib_ops/gqa.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
+# GroupQueryAttention — Operator Documentation
+
+This document describes the `com.microsoft::GroupQueryAttention` (GQA) contrib operator: its schema,
```
Comment (kunal-vaishnavi) [inline docs/contrib_ops/gqa.md:3]: Shall we move this doc inside `docs/contrib_ops/cuda/` instead?

**Review by kunal-vaishnavi — APPROVED** [review]
